OutOfMemory error in mirror Lambda function

Occurred on anvilprod and prod after the merge of the 2026-01-27 promotion (PR #7725 and PR #7726)

https://gitlab.explore.anvilproject.org/ucsc/azul/-/jobs/60484
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/91475

```
======================================================================
FAIL: test_indexing (integration_test.IndexingIntegrationTest.test_indexing) [mirror_sources_and_files]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 385, in subTest
    yield
  File "/builds/ucsc/azul/test/integration_test.py", line 1796, in _test_mirroring
    self._assert_queues_empty([config.mirror_queue.to_fail.name])
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/test/integration_test.py", line 1417, in _assert_queues_empty
    self.assertTrue(self.azul_client.is_queue_empty(queue_name))
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: False is not true
```

--- a/src/azul/terraform.py
+++ b/src/azul/terraform.py
@@ -1015,6 +1015,9 @@ class Chalice:
         # the runtime to 3.13v77, we observed a significant increase in memory
         # usage, leading to mirroring failures.
         #
+        # FIXME: Remove pinned Lambda runtime version
+        #        https://github.com/DataBiosphere/azul/issues/7798
+        #
         resource_type = 'aws_lambda_runtime_management_config'
         runtime_version = '1a6363019b274fc28ffbbec073e5ebf1a872c10c52269844a57c62e74063a49b'
         runtime_version_configs: MutableJSON = {}

--- a/src/azul/terraform.py
+++ b/src/azul/terraform.py
@@ -1011,6 +1011,23 @@ class Chalice:
             assert 'lifecycle' not in resource, (resource_name, resource)
             resource['lifecycle'] = {'create_before_destroy': True}
 
+        # Pin Lambda runtime to python:3.13.v76. When AWS automatically updated
+        # the runtime to 3.13v77, we observed a significant increase in memory
+        # usage, leading to mirroring failures.
+        #
+        resource_type = 'aws_lambda_runtime_management_config'
+        runtime_version = '1a6363019b274fc28ffbbec073e5ebf1a872c10c52269844a57c62e74063a49b'
+        runtime_version_configs: MutableJSON = {}
+        for resource_name, resource in resource_items('aws_lambda_function'):
+            runtime_version_configs[resource_name] = {
+                'function_name': '${aws_lambda_function.%s.function_name}' % resource_name,
+                'qualifier': '${aws_lambda_function.%s.version}' % resource_name,
+                'update_runtime_on': 'Manual',
+                'runtime_version_arn': 'arn:aws:lambda:us-east-1::runtime:' + runtime_version,
+            }
+        assert resource_type not in resources, resources
+        resources[resource_type] = runtime_version_configs
+
         return {
             'resource': resources,
             'data': data,

